### ISSUE TITLE
Add Collision group enum

### ISSUE THREAD

**Issue by dragokas** [alliedmodders/sourcemod#1618]
I think, it could be good idea to include this enum: https://developer.valvesoftware.com/wiki/Collision_groups in sdktools.inc as "int" type.
What do you think?

**Comment by sapphonie** [comment]
I think these should be included in game specific incs, e.g. tf2.inc, cstrike.inc, etc

**Comment by psychonic** [comment]
I don’t think an extra 20 inc files is the answer.

**Comment by sapphonie** [comment]
No, I mean for the games that have an existing inc, include the proper enum in there, for the games that do not, do not include a collision group enum.

**Comment by Kenzzer** [comment]
If you open the door for that, it won't prevent people from opening PRs.

SM is all or nothing regarding "generic engine feature". Furthermore the `cstrike` is used for both for CSS and CSGO and the collision enum is def not the same for the two games, so that idea doesn't work anyways.

**Comment by psychonic** [comment]
They can be consts in a shared inc, but the values need to be assigned at runtime (similar to MaxClients or the TFHoliday values).

**Comment by Kenzzer** [comment]
Would that be something that could be potentially added ? I thought the same regarding superseding sdkhooks, with pre configured hooks handle in consts, with the potential merge of #1629 .
That aside, considering I'm adding collision natives to sdktools lately, I'm up for making a PR regarding the addition of collision group consts.

**Comment by peace-maker** [comment]
Do some games have additional or less collision groups than others or are they always the same, just defined with different values? Having constants in the include that don't apply for your game seems confusing.

**Comment by psychonic** [comment]
> 
> 
> Do some games have additional or less collision groups than others or are they always the same, just defined with different values? Having constants in the include that don't apply for your game seems confusing.

I believe it's both. I wouldn't worry about ones that only exist in a single game. I like that to damage types in sdkhooks.inc that most games don't use half of.
